app/src/main/res/layout/main_fragment.xml

```diff
@@ -184,6 +184,46 @@
                 app:layout_constraintStart_toStartOf="@+id/start_guideline"
                 app:layout_constraintTop_toBottomOf="@+id/clear_session_button" />
 
+            <androidx.constraintlayout.widget.ConstraintLayout
+                android:id="@+id/visitor_code_switch_container"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                app:layout_constraintTop_toBottomOf="@id/visitor_code_button"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintEnd_toEndOf="parent" >
```

Comment:
NIT: `ConstraintLayout` provides all necessary mechanisms to not nest `ConstraintLayout`'s to each other. So this container is redundant. You can constraint views inside parent `ConstraintLayout` and use the `Group` widget to work with view groups of views e.g. change visibility etc.